ember-core/shared/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -62,26 +66,52 @@ private[ember] object Parser {
   )
 
   object HeaderP {
+    import scala.collection.mutable.ListBuffer
 
     private[this] val colon: Byte = ':'.toByte
     private[this] val contentLengthS = "Content-Length"
     private[this] val transferEncodingS = "Transfer-Encoding"
     private[this] val chunkedS = "chunked"
 
-    def parseHeaders[F[_]](message: Array[Byte], initIndex: Int, maxHeaderSize: Int)(implicit
+    final case class ParserState(
+        idx: Int,
+        state: Boolean,
+        throwable: Throwable,
+        complete: Boolean,
+        chunked: Boolean,
+        contentLength: Option[Long],
+        headers: ListBuffer[Header.Raw],
+        name: String,
+        start: Int,
+    )
+
+    object ParserState {
+      def initial: ParserState = ParserState(
```

Comment:
Why not just type them to options rather than introduce nulls?

Reply:
Mainly to make the change as non-invasive as possible. I can change it if you prefer?

Reply:
Yeah, its probably smoother if the state is that its Null or a value. There are combinators on Option to do that, which is useful as you can just convert it to the same types inside the function body, but be with good layers outside.
.orNull I think its called.